ksql-engine/src/main/java/io/confluent/ksql/util/ExecutorUtil.java

```diff
@@ -28,6 +31,9 @@
   private static final int NUM_RETRIES = 5;
   private static final Duration RETRY_BACKOFF_MS = Duration.ofMillis(500);
   private static final Logger log = LoggerFactory.getLogger(ExecutorUtil.class);
+  private static final Set<Class<? extends Throwable>> NO_RETRYABLE_EXCEPTIONS = ImmutableSet.of(
+      KsqlSchemaAuthorizationException.class
```

Comment:
Rather than add direct coupling here between `KsqlSchemaAuthorizationException` and the executor logic,  how about adding a `NotRetryableException` base class, which `KsqlSchemaAuthorizationException` can extend and which the executor can look for?